### DISCUSSION TITLE
Client Side error on Production build (NextJS)

### DISCUSSION THREAD

**Discussion by iqfareez** [scalar/scalar#732] [Q&A] [answered]
I followed the guide https://github.com/scalar/scalar/tree/main?tab=readme-ov-file#with-react to add scalar/api-reference to my Next Js app.

docs.tsx:
```ts
import { ApiReference as VueComponent } from '@scalar/api-reference'
import { applyVueInReact } from 'veaury'

const ApiReference = applyVueInReact(VueComponent)

function App() {
    const configuration = {
        spec: {
            url: 'https://raw.githubusercontent.com/mptwaktusolat/open-api-spec/main/mpt-server-api.yaml'
        }
    };
    return (
        <>
            {/* @ts-ignore */}
            <ApiReference configuration={configuration} />
        </>
    )
}

export default App
```

Error `Application error: a client-side exception has occurred (see the browser console for more information).` shown.

<img width="1440" alt="Screenshot 2024-01-02 at 10 05 48 AM" src="https://github.com/scalar/scalar/assets/60868965/1899e60c-f4a5-4b81-bc95-a60d928abefe">

![Screenshot 2024-01-02 at 10 15 54 AM](https://github.com/scalar/scalar/assets/60868965/dda55c35-ae99-498c-9ea8-3166fafe60ce)

Perhaps it is somethiing to do with the `__VUE_PROD_DEVTOOLS__` variable. Can anyone help on how to make this work.

Note: This issue didn't occur on development (`next dev`)

Version:
`"@scalar/api-reference": "^1.12.7"`

**Comment by iqfareez** [comment, marked as answer]
Solved.

Looks like vue just released a [major version](https://github.com/vuejs/core/blob/main/CHANGELOG.md#340-slam-dunk-2023-12-29) few days ago

'Pin' the dependencies to the older vue to force scalar use that one.

Add in package.json:

```json
  "resolutions": {
    "vue" :"3.3.13"
  }
```

**Reply by hanspagel** [reply]
Thanks for sharing the workaround!
